File: 1-Text-Based/C/pemilu-2024.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int main() {
    char *paslonPemilu2024[] = {"Anies-Muhaimin", "Prabowo-Gibran", "Ganjar-Mahfud"};
    char *parpolPemilu2024[] = {"PKB", "Gerindra", "PDIP", "Golkar", "Nasdem", "Buruh", "Gelora", "PKS", "PKN", "Hanura", "Garuda", "PAN", "PBB", "Demokrat", "PSI", "Perindo", "PPP", "Ummat"};

    while (true) {
        printf("\n======================================\n");
        printf("     PROGRAM PEMILIHAN UMUM 2024     \n");
        printf("--------------------------------------\n");
        printf("1. Tampilkan Nama Paslon dan Partai\n");
        printf("2. Pilih Paslon dan Partai\n");
        printf("======================================\n");

        int choice;
        printf("Masukkan pilihan Anda : ");
        scanf("%d", &choice);

        if (choice == 1) {
            printf("\n======================================\n");
            printf("Pasangan Calon :\n");
            for (int i = 0; i < sizeof(paslonPemilu2024) / sizeof(paslonPemilu2024[0]); ++i) {
                printf("%d. %s\n", i + 1, paslonPemilu2024[i]);
            }

            printf("\nPartai Politik :\n");
            for (int i = 0; i < sizeof(parpolPemilu2024) / sizeof(parpolPemilu2024[0]); ++i) {
                printf("%d. %s\n", i + 1, parpolPemilu2024[i]);
            }
            printf("======================================\n");
        } else if (choice == 2) {
            int paslon_pilihan;
            printf("\nMasukkan Nomor Paslon yang dipilih : ");
            scanf("%d", &paslon_pilihan);
            if (!(1 <= paslon_pilihan && paslon_pilihan <= sizeof(paslonPemilu2024) / sizeof(paslonPemilu2024[0]))) {
                printf("Nomor paslon tidak valid.\n");
                continue;
            }

            char *partai_koalisi[10]; // Tentukan ukuran maksimal array partai_koalisi
            if (strcmp(paslonPemilu2024[paslon_pilihan - 1], "Anies-Muhaimin") == 0) {
                char *temp[] = {"PKB", "Nasdem", "PKS", "Ummat"};
                for (int i = 0; i < sizeof(temp) / sizeof(temp[0]); ++i) {
                    partai_koalisi[i] = temp[i];
                }
            } else if (strcmp(paslonPemilu2024[paslon_pilihan - 1], "Ganjar-Mahfud") == 0) {
                char *temp[] = {"PDIP", "Hanura", "Perindo", "PPP"};
                for (int i = 0; i < sizeof(temp) / sizeof(temp[0]); ++i) {
                    partai_koalisi[i] = temp[i];
                }
            } else if (strcmp(paslonPemilu2024[paslon_pilihan - 1], "Prabowo-Gibran") == 0) {
                char *temp[] = {"Gerindra", "Golkar", "Buruh", "Gelora", "PKN", "Garuda", "PAN", "PBB", "Demokrat", "PSI"};
                for (int i = 0; i < sizeof(temp) / sizeof(temp[0]); ++i) {
                    partai_koalisi[i] = temp[i];
                }
            } else {
                for (int i = 0; i < sizeof(parpolPemilu2024) / sizeof(parpolPemilu2024[0]); ++i) {
                    partai_koalisi[i] = parpolPemilu2024[i];
                }
            }

            printf("\n======================================\n");
            printf("Partai Politik Koalisi %s :\n", paslonPemilu2024[paslon_pilihan - 1]);
            for (int i = 0; i < sizeof(partai_koalisi) / sizeof(partai_koalisi[0]); ++i) {
                if (partai_koalisi[i] != NULL) { // Periksa apakah ada partai yang tidak null
                    printf("%d. %s\n", i + 1, partai_koalisi[i]);
                }
            }
            printf("======================================\n");

            int partai_pilihan;
            printf("\nMasukkan Nomor Partai yang dipilih : ");
            scanf("%d", &partai_pilihan);
            if (!(1 <= partai_pilihan && partai_pilihan <= sizeof(partai_koalisi) / sizeof(partai_koalisi[0]))) {
                printf("Nomor partai tidak valid.\n");
                continue;
            }

            printf("\n======================================\n");
            printf("             PILIHAN ANDA             \n");
            printf("--------------------------------------\n");
            printf("Pasangan Calon : %s\n", paslonPemilu2024[paslon_pilihan - 1]);
            printf("Partai Politik : %s\n", partai_koalisi[partai_pilihan - 1]);
            printf("======================================\n");
        } else {
            printf("Pilihan tidak valid. Silakan masukkan 1 atau 2.\n");
            continue;
        }

        char exit_choice;
        printf("\nApakah Anda ingin keluar? (y/n): ");
        scanf(" %c", &exit_choice);
        if (exit_choice == 'y' || exit_choice == 'Y') {
            break;
        }
    }

    printf("\n");

    return 0;
}

File: 1-Text-Based/Java/Pemilu2024.java
import java.util.Scanner;

public class Pemilu2024 {
    private static final String[] paslonPemilu2024 = {"Anies-Muhaimin", "Prabowo-Gibran", "Ganjar-Mahfud"};
    private static final String[] parpolPemilu2024 = {"PKB", "Gerindra", "PDIP", "Golkar", "Nasdem", "Buruh", "Gelora", "PKS", "PKN", "Hanura", "Garuda", "PAN", "PBB", "Demokrat", "PSI", "Perindo", "PPP", "Ummat"};

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\n======================================");
            System.out.println("     PROGRAM PEMILIHAN UMUM 2024     ");
            System.out.println("--------------------------------------");
            System.out.println("1. Tampilkan Nama Paslon dan Partai");
            System.out.println("2. Pilih Paslon dan Partai");
            System.out.println("======================================");

            System.out.print("Masukkan pilihan Anda : ");
            int choice = scanner.nextInt();

            if (choice == 1) {
                System.out.print("\n======================================");
                System.out.println("\nPasangan Calon :");
                for (int i = 0; i < paslonPemilu2024.length; i++) {
                    System.out.println((i + 1) + ". " + paslonPemilu2024[i]);
                }

                System.out.println("\nPartai Politik :");
                for (int i = 0; i < parpolPemilu2024.length; i++) {
                    System.out.println((i + 1) + ". " + parpolPemilu2024[i]);
                }
                System.out.println("======================================");
            } else if (choice == 2) {
                System.out.print("\nMasukkan Nomor Paslon yang dipilih : ");
                int paslon_pilihan = scanner.nextInt();
                if (!(1 <= paslon_pilihan && paslon_pilihan <= paslonPemilu2024.length)) {
                    System.out.println("Nomor paslon tidak valid.");
                    continue;
                }

                String[] partai_koalisi;
                if (paslonPemilu2024[paslon_pilihan - 1].equals("Anies-Muhaimin")) {
                    partai_koalisi = new String[]{"PKB", "Nasdem", "PKS", "Ummat"};
                } else if (paslonPemilu2024[paslon_pilihan - 1].equals("Ganjar-Mahfud")) {
                    partai_koalisi = new String[]{"PDIP", "Hanura", "Perindo", "PPP"};
                } else if (paslonPemilu2024[paslon_pilihan - 1].equals("Prabowo-Gibran")) {
                    partai_koalisi = new String[]{"Gerindra", "Golkar", "Buruh", "Gelora", "PKN", "Garuda", "PAN", "PBB", "Demokrat", "PSI"};
                } else {
                    partai_koalisi = parpolPemilu2024;
                }

                System.out.println("\n======================================");
                System.out.println("Partai Politik Koalisi " + paslonPemilu2024[paslon_pilihan - 1] + " :");
                for (int i = 0; i < partai_koalisi.length; i++) {
                    System.out.println((i + 1) + ". " + partai_koalisi[i]);
                }
                System.out.print("======================================");

                System.out.print("\nMasukkan Nomor Partai yang dipilih : ");
                int partai_pilihan = scanner.nextInt();
                if (!(1 <= partai_pilihan && partai_pilihan <= partai_koalisi.length)) {
                    System.out.println("Nomor partai tidak valid.");
                    continue;
                }

                System.out.println("\n======================================");
                System.out.println("             PILIHAN ANDA             ");
                System.out.println("--------------------------------------");
                System.out.println("Pasangan Calon : " + paslonPemilu2024[paslon_pilihan - 1]);
                System.out.println("Partai Politik : " + partai_koalisi[partai_pilihan - 1]);
                System.out.println("======================================");
            } else {
                System.out.println("Pilihan tidak valid. Silakan masukkan 1 atau 2.");
                continue;
            }

            System.out.print("\nApakah Anda ingin keluar? (y/n): ");
            String exit_choice = scanner.next().toLowerCase();
            if (exit_choice.equals("y")) {
                break;
            }
        }

        System.out.println("\n");
    }
}
